src/components/AudioPlayer/HighlightedLocationUpdater.tsx

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useRef } from 'react';
+
+import { shallowEqual, useDispatch } from 'react-redux';
+import useSWRImmutable from 'swr/immutable';
+
+import getHighlightedLocation from './getHighlightedLocation';
+
+import { getChapterAudioFile } from 'src/api';
+import {
+  initialState as defaultHighlightedLocation,
+  setHighlightedLocation,
+} from 'src/redux/slices/QuranReader/highlightedLocation';
+import { makeChapterAudioFiles } from 'src/utils/apiPaths';
+
+type AudioTimestampsHighlightListenerProps = {
+  reciterId: number;
+  chapterId: number;
+  currentTime: number;
+};
+
+/**
+ * Listen to audio player's currentTime and update highlightedLocation state
+ *
+ * @returns {void}
+ */
+const HighlightedLocationUpdater = ({
+  reciterId,
+  chapterId,
+  currentTime,
+}: AudioTimestampsHighlightListenerProps) => {
+  const dispatch = useDispatch();
+  const { data } = useSWRImmutable(makeChapterAudioFiles(reciterId, chapterId, true), () =>
+    getChapterAudioFile(reciterId, chapterId, true),
+  );
+
+  const lastHighlightedLocation = useRef(defaultHighlightedLocation);
+
+  // eslint-disable-next-line consistent-return
+  useEffect(() => {
+    if (!data) return;
+
+    const highlightedLocation = getHighlightedLocation(
+      window.audioPlayerEl.currentTime * 1000, // convert currentTime to milliseconds
+      data.verseTimings,
+    );
+
+    // only update redux value if current highlight value does not equal prev value
+    if (!shallowEqual(highlightedLocation, lastHighlightedLocation.current)) {
+      dispatch(setHighlightedLocation(highlightedLocation));
+      lastHighlightedLocation.current = highlightedLocation;
+    }
+  }, [data, dispatch, currentTime]);
+
+  return null;
```

Comment:
no need to return null here

Reply:
I think React will be angry if we don't return null here?

![image](https://user-images.githubusercontent.com/12745166/134129239-e04e6d50-e977-4676-b2e1-d349f1c8f81b.png)